eras/conway/impl/src/Cardano/Ledger/Conway/Rules/GovCert.hs

```diff
@@ -164,48 +167,57 @@ conwayGovCertTransition ::
   (ConwayEraPParams era, EraRule "GOVCERT" era ~ ConwayGOVCERT era) => TransitionRule (ConwayGOVCERT era)
 conwayGovCertTransition = do
   TRC
-    ( ConwayGovCertEnv {..}
+    ( ConwayGovCertEnv {cgcePParams, cgceCurrentSlot}
       , vState@VState {vsDReps, vsCommitteeHotKeys}
       , c
       ) <-
     judgmentContext
   case c of
-    ConwayRegDRep cred _deposit mAnchor -> do
-      Map.notMember cred vsDReps ?! ConwayDRepAlreadyRegistered cred
-      -- TODO: check against a new PParam `drepDeposit`, once PParams are updated. -- someCheck ?! ConwayDRepIncorrectDeposit deposit
-      drepState <- do
-        expiryEpoch <-
-          calcDRepExpiryEpoch
-            cgceCurrentSlot
-            (cgcePParams ^. ppDRepActivityL)
-        pure
-          DRepState
-            { drepExpiry = expiryEpoch
-            , drepAnchor = mAnchor
-            }
+    -- GOVCERT-regdrep : let open PParams pp in
+    --   (d ≡ drepDeposit × c ∉ dom (dReps ˢ)) ⊎ (d ≡ 0 × c ∈ dom (dReps ˢ))
+    --   ────────────────────────────────
+    --   ⟦ e , pp , vs ⟧ᶜ ⊢ ⟦ dReps , ccKeys ⟧ᵛ ⇀⦇ regdrep c d an ,GOVCERT⦈
+    --                      ⟦ ❴ c , e +ᵉ' drepActivity ❵ᵐ ∪ᵐˡ dReps , ccKeys ⟧ᵛ
+    ConwayRegDRep cred deposit mAnchor -> do
+      (cgcePParams ^. ppDRepDepositL == deposit && Map.notMember cred vsDReps)
+        || (Coin 0 == deposit && Map.member cred vsDReps)
+          ?! ConwayDRepRegMalformedRequest deposit cred
+      expiryEpoch <- calcDRepExpiryEpoch cgceCurrentSlot $ cgcePParams ^. ppDRepActivityL
+      let drepState = DRepState expiryEpoch mAnchor
       pure $ vState {vsDReps = Map.insert cred drepState vsDReps}
```

Comment:
I think we should leave this as it was before, because we decided to handle the right side of `||` using the `ConwayUpdateDRep` certificate.

Reply:
Ah! Thank you! :raised_hands: